### PR TITLE
Add support solargraph

### PR DESCRIPTION
**Pull request by h-michael**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -906,6 +907,129 @@ nvim_lsp#setup("rls", {config})
     settings = {}
 ```
 
+## solargraph
```
Comment (justinmk) [inline README.md]: this should go in `docs.description` i think? README.md is updated by a GH Action.

Comment (h-michael) [reply README.md]: I executed `sh ./scripts/docgen.sh`.
(https://github.com/neovim/nvim-lsp/blob/ac3df47ffd418b0f3aec49612d9862364820ed8d/CONTRIBUTING.md)

So, if we change the format, we should change https://github.com/neovim/nvim-lsp/blob/0d30787a47b4f4d04d5a08fe6eadf309c8e055b7/scripts/docgen.lua .

**Comment by mikker** [comment]
I'm struggling to get diagnostics working. Can someone post a sample configuration with diagnostics enabled?

**Comment by justinmk** [comment]
@mikker what did you try? Assuming you have the server installed, diagnostics should appear with just this line in your config (+Nvim HEAD, +nvim-lsp HEAD):

    require'nvim_lsp'.solargraph.setup{}

If not, check the log file for errors:

    :exe 'edit' v:lua.vim.lsp.get_log_path()

**Comment by mikker** [comment]
Log saying this

```
[ ERROR ] 2019-12-11T15:27:07Z+0100 ] ...ovim/HEAD-d9ab83d/share/nvim/runtime/lua/vim/lsp/rpc.lua:315 ]	"rpc"	"solargraph"	"stderr"	"Solargraph is listening on stdio PID=46976\n"
[ ERROR ] 2019-12-11T15:27:13Z+0100 ] ...ovim/HEAD-d9ab83d/share/nvim/runtime/lua/vim/lsp/rpc.lua:315 ]	"rpc"	"solargraph"	"stderr"	"[ANY] Solargraph initialized (4.750450000021374 seconds)\n"
[ ERROR ] 2019-12-11T15:33:43Z+0100 ] ...ovim/HEAD-d9ab83d/share/nvim/runtime/lua/vim/lsp/rpc.lua:315 ]	"rpc"	"solargraph"	"stderr"	"/Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/client.rb:73:in `synchronize': can't be called from trap context (ThreadError)\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/client.rb:73:in `read'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/client.rb:62:in `tick'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/server/stdio.rb:18:in `update'\n\tfrom /Users/mikker/.rubies/ruby-2.6.5/lib/ruby/2.6.0/observer.rb:197:in `block in notify_observers'\n\tfrom /Users/mikker/.rubies/ruby-2.6.5/lib/ruby/2.6.0/observer.rb:196:in `each'\n\tfrom /Users/mikker/.rubies/ruby-2.6.5/lib/ruby/2.6.0/observer.rb:196:in `notify_observers'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/client.rb:41:in `stop'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/server/connectable.rb:13:in `map'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/server/connectable.rb:13:in `stopping'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/server/base.rb:23:in `stop'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/machine.rb:26:in `map'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/machine.rb:26:in `stop'\n"
[ ERROR ] 2019-12-11T15:33:43Z+0100 ] ...ovim/HEAD-d9ab83d/share/nvim/runtime/lua/vim/lsp/rpc.lua:315 ]	"rpc"	"solargraph"	"stderr"	"\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport.rb:58:in `stop'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/solargraph-0.38.0/lib/solargraph/language_server/transport/adapter.rb:54:in `shutdown'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/solargraph-0.38.0/lib/solargraph/language_server/transport/adapter.rb:35:in `update'\n\tfrom /Users/mikker/.rubies/ruby-2.6.5/lib/ruby/2.6.0/observer.rb:197:in `block in notify_observers'\n\tfrom /Users/mikker/.rubies/ruby-2.6.5/lib/ruby/2.6.0/observer.rb:196:in `each'\n\tfrom /Users/mikker/.rubies/ruby-2.6.5/lib/ruby/2.6.0/observer.rb:196:in `notify_observers'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/solargraph-0.38.0/lib/solargraph/language_server/host.rb:444:in `stop'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/solargraph-0.38.0/lib/solargraph/language_server/transport/adapter.rb:22:in `closing'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/client.rb:38:in `stop'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/server/connectable.rb:13:in `map'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/server/connectable.rb:13:in `stopping'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/server/base.rb:23:in `stop'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/machine.rb:26:in `map'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/machine.rb:26:in `stop'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport.rb:58:in `stop'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/solargraph-0.38.0/lib/solargraph/shell.rb:44:in `block (2 levels) in stdio'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/machine.rb:75:in `sleep'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/machine.rb:75:in `run_server_thread'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport/machine.rb:19:in `run'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/backport-1.1.2/lib/backport.rb:51:in `run'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/solargraph-0.38.0/lib/solargraph/shell.rb:39:in `stdio'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/gems/solargraph-0.38.0/bin/solargraph:5:in `<top (required)>'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/bin/solargraph:23:in `load'\n\tfrom /Users/mikker/.gem/ruby/2.6.5/bin/solargraph:23:in `<main>'\n"
```

**Comment by tjaartvdwalt** [comment]
@mikker I know this is an old thread, but I wanted to post my findings for future readers.

I found that diagnostics does not seem to be enabled by default, but I did get it to work when I manually enabled the setting in the setup function

```lua
local lsp = require 'nvim_lsp'
lsp.solargraph.setup{
  settings = {
    solargraph = {
      diagnostics = true
    }
  }
}

```

**Pull request merged**